### PR TITLE
Dynamic graph api versioning

### PR DESCRIPTION
**Pull request by pedroseabra1091**
## Ticket
https://www.pivotaltracker.com/story/show/180464208

## Changelog
- Dynamic version for each of the endpoints (`bot`, `labels`, `profile`, `personas`, `subscriptions`). Each resource uses the config provider to either fetch an injected graph API version or the default version defined on `environment.rb`;
- Fixes to a few outdated specs.

### CODE REVIEW

File: lib/facebook/messenger/bot.rb [hunk 1]
```diff
@@ -171,6 +171,15 @@ def default_options
...
         end
+
+        #
+        # Returns the graph api version defined in the provider.
+        #
+        # @return [String] graph api version.
+        #
+        def graph_api_version
+          Facebook::Messenger.config.provider.graph_api_version
+        end
```
Comment (pedroseabra1091) [inline lib/facebook/messenger/bot.rb:182]: Each resource has one method like this, to the eventuality that we need to lock a resource to a specific version.

File: lib/facebook/messenger/configuration/providers/environment.rb [hunk 2]
```diff
@@ -27,8 +29,8 @@ def fetch_app_secret_proof_enabled?
             ENV['APP_SECRET_PROOF_ENABLED'] == 'true'.freeze
           end
 
-          def fb_graph_api_version
-            ENV['FB_GRAPH_API_VERSION']
+          def graph_api_version
+            GRAPH_API_VERSION
```
Comment (taniasoaressantos) [inline lib/facebook/messenger/configuration/providers/environment.rb]: Why are we defining this as a constant?

Comment (pedroseabra1091) [reply lib/facebook/messenger/configuration/providers/environment.rb]: This is just a measure to ensure that if a version isn't injected, there's a fallback in place.
In our case, we override this environment with `BotConfigurator`, but if that weren't to happen, we would have this version to fallback to.

**Review by taniasoaressantos — APPROVED** [review]

**Review by anam-mansoor7 — APPROVED** [review]

**Comment by anam-mansoor7** [comment]
Nice one 🚀

**Pull request merged**